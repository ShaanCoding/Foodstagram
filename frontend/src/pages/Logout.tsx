import Cookies from 'js-cookie'
import { Navigate } from 'react-router-dom'

const Logout = () => {
	if (Cookies.get('access_token')) {
		Cookies.set('access_token', '')
	}

	return <Navigate replace to="/login" />
}

export default Logout
