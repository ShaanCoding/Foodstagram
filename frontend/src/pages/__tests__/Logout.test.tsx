import { QueryClient, QueryClientProvider } from 'react-query'
import { BrowserRouter, Router } from 'react-router-dom'
import renderer from 'react-test-renderer'
import Logout from '../Logout'

const queryClient = new QueryClient()

it('The Logout page renders correctly', () => {
	const tree = renderer
		.create(
			<QueryClientProvider client={queryClient}>
				<BrowserRouter>
					<Logout />
				</BrowserRouter>
			</QueryClientProvider>
		)
		.toJSON()
	expect(tree).toMatchSnapshot()
})
