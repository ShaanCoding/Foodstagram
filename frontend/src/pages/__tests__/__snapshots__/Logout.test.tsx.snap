// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The Logout page renders correctly 1`] = `
<div>
  Logging out...
</div>
`;
